templates/new/web_dodo.py.txt

```diff
@@ -148,6 +157,286 @@ def task_web_local():
+
+    # Get configuration from global/local config
+    global_config = get_surround_config()
+    user_name = global_config.get_path("user.name")
+    user_email = global_config.get_path("user.email")
+    experiment_url = global_config.get_path("experiment.url")
+    
+    credentials = compute_get_credentials()
+    if not credentials:
+        # Fail task since we have no credentials
+        return "", lambda: False
+
+    # Copy the credentials file to the root of the project
+    with open(os.environ["GOOGLE_APPLICATION_CREDENTIALS"], "rb") as cf:
+        with open("credentials.json", "wb+") as tf:
+            tf.write(cf.read())
+
+    # Construct image uri
+    image_uri = "gcr.io/%s/%s" % (credentials.project_id, IMAGE if not is_gpu else "%s-gpu" % IMAGE)
+
+    # Construct build command (and clean-up)
+    build_cmd = "docker build -f %s " % dockerfile
+    build_cmd += "--build-arg \"EXPERIMENT_URL=%s\" " % experiment_url
+    build_cmd += "--build-arg \"USER_NAME=%s\" " % user_name
+    build_cmd += "--build-arg \"USER_EMAIL=%s\" " % user_email
+    build_cmd += "-t %s " % image_uri
+    build_cmd += "./"
+
+    return image_uri, build_cmd
+
+def task_compute_build():
+    """Build compute image and push to Google Cloud (Container Registry)"""
+
+    # Construct build command
+    image_uri, build_cmd = compute_docker_build_cmd("Dockerfile-compute")
+    
+    # Construct push command
+    push_cmd = "docker push %s" % image_uri
+
+    return {{
+        'basename': 'buildCompute',
+        'actions': [compute_docker_login, build_cmd, 'rm credentials.json', push_cmd]
+    }}
+
+def task_compute_build_gpu():
+    """Build GPU compute image and push to Google Cloud (Container Registry)"""
+
+    # Construct build command
+    image_uri, build_cmd = compute_docker_build_cmd("Dockerfile-compute-gpu", True)
+    
+    # Construct push command
+    push_cmd = "docker push %s" % image_uri
+
+    return {{
+        'basename': 'buildComputeGPU',
+        'actions': [compute_docker_login, build_cmd, 'rm credentials.json', push_cmd]
+    }}
+
+def compute_check_image(is_gpu=False):
+    """Checks whether there has been an image build for compute"""
+
+    image_name = IMAGE if not is_gpu else ("%s-gpu" % IMAGE)
+    process = subprocess.Popen(['docker', 'images', '-q', '-f', 'reference=gcr.io/*/%s' % image_name], stdout=subprocess.PIPE)
+    process.wait()
+
+    if not process.stdout.read():
+        print("No compute image found!")
+        print("Make sure you run the following before deploying to the cloud!")
+        print("$ surround run %s" % ("buildComputeGPU" if is_gpu else "buildCompute"))
+        return False
+
+def compute_deploy_job(mode="batch", is_gpu=False):
+    """Deploys a compute job to Google Cloud (AI Platform)"""
+
+    credentials = compute_get_credentials()
+    if not credentials:
+        return False
+
+    # python api for AI Platform
+    cloud_ml = discovery.build('ml', 'v1', credentials=credentials)
```

Comment:
Intermittently getting ssl.SSLEOFError from this line. Would suggest catching the error and retrying the request as each time I tried the same command directly afterwards and it worked. Should be mirrored every time discovery.build is called in this and the non-web version. Attached image of error occurring.
![image](https://user-images.githubusercontent.com/12322598/68730165-06bd2280-0620-11ea-97d6-7b843e50364b.png)